pkg/uevent/queue.go

```diff
@@ -68,42 +69,25 @@ func newEventQueue() *eventQueue {
 //   Update()
 //   Delete()
 // }
-// 
+//
 // pkg/udev/uevent/listener.go
 // ev := q.pop()
 // if err := handle(ev); err != nil {
 //   q.push(ev)
 // }
-// 
-func (e *eventQueue) push(event *deviceEvent) {
+//
+
+func (e *eventQueue) set(event *deviceEvent) (existingEvent *deviceEvent, found bool) {
 	e.mutex.Lock()
 	defer e.mutex.Unlock()
 
-	addBackoff = event.popped
-
-	existingEvent, found := e.events[event.devPath]
+	existingEvent, found = e.events[event.devPath]
 	if found {
```

Comment:
The `else if` block will never be executed if the existing event is found with proper order. 

We can change to this,

```suggestion
	if found && existingEvent.created.After(event.created) {
	     // Skip as passed event is older.
	     return nil, found
        }
```

Reply:
If we do that, we miss adding newer events for the same key.